tools/aws_lab_setup/teardown_lab.yml

```diff
@@ -2,17 +2,46 @@
   hosts: localhost
   connection: local
   become: no
-  gather_facts: yes
+  gather_facts: no
 
   vars:
-    ec2_exact_count: 0
     ec2_wait: no
+    teardown: true
+    create_lab: false
+    s3_state: absent
+    state: absent
+
+  pre_tasks:
+  - name: Include dynamically created generated_student_list.txt
+    include_vars:
+      file: '{{ playbook_dir }}/{{ec2_name_prefix}}_generated_student_list.txt'
+    when: (users is undefined or not users) and (student_total is undefined or not student_total)
 
   roles:
     - manage_ec2_instances
+    - { role: aws_workshop_login_page, when: create_login_page is defined }
 
   post_tasks:
-    - name: Remove inventory files
-      shell: rm -vf {{ playbook_dir }}/*.txt
-      register: removal
-      changed_when: removal.stdout_lines | length > 0
+   - name: Remove inventory files
+     file:
+       dest: "{{item}}"
+       state: absent
+     with_fileglob:
+       - "{{ playbook_dir }}/*.txt"
+
+   - name: Remove HTML files
+     file:
+       dest: "{{item}}"
+       state: absent
+     with_fileglob:
+       - "{{ playbook_dir }}/*.html"
+
+   - name: Remove private key file
+     file:
+       path: '{{ playbook_dir }}/*.pem'
+       state: absent
+
+   - name: Remove old keys
+     shell: ssh-add -D
```

Comment:
Use the `authorized_key` module not shell. (Best practice is that this should have been a `command` and not a `shell` anyways.)

Reply:
@tima that module is for public keys, I need to remove the private key since it was a temporary private/public key just for the workshop.

What will happen is if you run the provisioner a bunch of times without specifying a key there will be timeouts trying to connect b/c there will be too many private keys to shuffle through.

Is there another module for removing private keys or should I just change to `command`

Reply:
Just change to `command` for now.